### PR TITLE
Add success policy to TF training job

### PR DESCRIPTION
**Pull request by ChenYi015**
## Purpose of this PR

**Proposed changes:**

- Add success policy to TF training job

## Change Category

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

### CODE REVIEW

File: pkg/argsbuilder/submit_tfjob.go [hunk 1]
```diff
@@ -130,6 +130,8 @@ func (s *SubmitTFJobArgsBuilder) AddCommandFlags(command *cobra.Command) {
 	command.Flags().StringVar(&s.args.PSMemory, "ps-memory", "", "the memory resource to use for the parameter servers, like 1Gi.")
 	command.Flags().StringVar(&s.args.PSMemoryLimit, "ps-memory-limit", "", "the memory resource limit to use for the parameter servers, like 1Gi.")
 
+	command.Flags().StringVar(&s.args.SuccessPolicy, "success-policy", "", "Specifies the policy to mark the TFJob as succeeded.")
```
Comment (cheyang) [inline pkg/argsbuilder/submit_tfjob.go:133]: Please indicate what policy is supported and what policy is default.  You can take a look at 

```golang
command.Flags().StringVar(&s.args.CleanPodPolicy, "cleanTaskPolicy", "Running", "How to clean tasks after Training is done, support Running, None and All.")
```

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/argsbuilder/submit_tfjob.go (hunk 1 above)
Comment (ChenYi015) [reply pkg/argsbuilder/submit_tfjob.go]: Done.

File: arena-artifacts/all_crds/v1/tf-operator/kubeflow.org_tfjobs_v1.yaml [hunk 2]
```diff
@@ -66,6 +66,11 @@ spec:
                   a positive integer.
                 format: int64
                 type: integer
+              successPolicy:
```
Comment (happy2048) [inline arena-artifacts/all_crds/v1/tf-operator/kubeflow.org_tfjobs_v1.yaml:69]: please update following annotation:  https://github.com/kubeflow/arena/blob/master/arena-artifacts/all_crds/v1/tf-operator/kubeflow.org_tfjobs_v1.yaml#L9

File: charts/tfjob/templates/tfjob.yaml [hunk 3]
```diff
@@ -52,6 +52,9 @@ spec:
   ttlSecondsAfterFinished: {{ .Values.ttlSecondsAfterFinished }}
   {{- end }}
 {{- end }}
+  {{- with .Values.successPolicy }}
+  successPolicy: {{ . }}
```
Comment (happy2048) [inline charts/tfjob/templates/tfjob.yaml:56]: successPolicy: {{ .Values.successPolicy }}

File: pkg/argsbuilder/submit_tfjob.go [hunk 4]
```diff
@@ -130,6 +130,8 @@ func (s *SubmitTFJobArgsBuilder) AddCommandFlags(command *cobra.Command) {
 	command.Flags().StringVar(&s.args.PSMemory, "ps-memory", "", "the memory resource to use for the parameter servers, like 1Gi.")
 	command.Flags().StringVar(&s.args.PSMemoryLimit, "ps-memory-limit", "", "the memory resource limit to use for the parameter servers, like 1Gi.")
 
+	command.Flags().StringVar(&s.args.SuccessPolicy, "success-policy", "", "Specifies the policy to mark the TFJob as succeeded. Available options are \"\" and AllWorkers. Default to \"\".")
```
Comment (happy2048) [inline pkg/argsbuilder/submit_tfjob.go:133]: Available Value: AllWorkers and Chief？

Comment (cheyang) [reply pkg/argsbuilder/submit_tfjob.go:133]: Yes, I think need clarify what default value indicates.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**